spark/src/test/scala/org/apache/spark/sql/delta/test/DeltaSQLTestUtils.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.test
+
+import java.io.File
+
+import org.apache.spark.sql.test.SQLTestUtils
+import org.apache.spark.util.Utils
+
+trait DeltaSQLTestUtils extends SQLTestUtils {
+  /**
+   * Generates a temporary path with a prefix without creating the actual file/directory,
+   * and passes it to `f`. If a file/directory is created by `f`, it will be deleted
+   * after `f` returns.
+   *
+   * This method is copied over from [[SQLTestUtils]] of Apache Spark.
+   */
+  def withTempDir(prefix: String)(f: File => Unit): Unit = {
+    val path = Utils.createTempDir(namePrefix = prefix)
+    path.delete()
```

Comment:
What is the purpose of this `.delete()` call? Did we meant to do it here or part of the `finally` clause? I don't see `delete` in the Spark implementation. https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala#L78

Reply:
I copied from here: https://src.dev.databricks.com/databricks/runtime@master/-/blob/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/SQLHelper.scala?L41:5

Apparently there are multiple implementations... alright now I'll copy from SQLTestUtils.scala instead.